invenio_previewer/extensions/ipynb.py

```diff
@@ -24,10 +25,12 @@ def render(file):
     """Generate the result HTML."""
     with file.open() as fp:
         content = fp.read()
+    try:
+        notebook = nbformat.reads(content.decode("utf-8"), as_version=4)
+    except nbformat.reader.NotJSONError:
+        return _("Error: Not a json file"), {}
 
-    notebook = nbformat.reads(content.decode("utf-8"), as_version=4)
-
-    html_exporter = HTMLExporter()
+    html_exporter = HTMLExporter(template="lab", embed_images=True)
```

Comment:
If we can trust the html created by nbformat, we could remove the `sanitize_html` from the template. That helps in two things:
* The sanitize_html removes the embedded plots
* the sanitize_html removes part of the formatting, in particular the `red` around the error of the commands.

Reply:
Unfortunately I don't think that's possible. These files are user-uploaded so we must be very strict with what comes out of it. We previously had an issue with ipynb allowing XSS injection so it would require a quite deep review of nbconvert, or that files are served from a different domain name.

However, it is possible to customise ``sanitize_html`` to perhaps allow these specific things.

Reply:
See 
- https://github.com/inveniosoftware/invenio-formatter/blob/master/invenio_formatter/filters/html.py#L15-L30C14
- https://github.com/inveniosoftware/invenio-config/blob/master/invenio_config/default.py#L16-L51

So something like 
```python
sanitize_html(tags=config["ALLOWED_HTML_TAGS_IPYNB"], attributes=config["ALLOWED_HTML_ATTRIBUTES_IPYNB"])
```